bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -738,6 +627,159 @@ public Image(Device device, ImageDataProvider imageDataProvider) {
 	init();
 }
 
+
+/**
+ * Upon zoom update, this method copies the image data and add disabled or grayed image depending on the flag
+ *
+ * @param device the device on which to create the image
+ * @param data the image data to create the image from (must not be null)
+ * @param flag the style, either <code>IMAGE_DISABLE</code> or <code>IMAGE_GRAY</code>
+ * @return new image data with valid styling properties as per the flag
+ */
+private ImageData copyImageDataForDisabledOrGrayImages(Device device, ImageData data, int flag) {
```

Comment:
I found the changes hard to comprehend at first sight, because the method does not do what it's name suggests: it does not copy data (only) for disabled or gray images, but transforms the data in these two cases and simply returns the original data in other cases. So maybe this should be somethink like `adaptImageDataIfDisabledOrGray()`?

Reply:
Yes, that's more clear.